### PR TITLE
Change supported Elixir and OTP versions

### PR DESCRIPTION
**Pull request by philss**
The idea is to restrict the versions we support to be the latest 3
versions of Elixir and OTP.
So right now we support Elixir ~> 1.11 and OTP ~> 22.

### CODE REVIEW

**Review by hansihe — APPROVED** [review]
LGTM

**Comment by evnu** [comment]
Currently, the README states:

> Rustler aims to support the newest three OTP versions as well as Elixir versions capable of running the supported versions of OTP.

This will need to be adapted as well if we also restrict the supported Elixir versions.

**Comment by philss** [comment]
@evnu good point! I changed the text.

File: rustler_tests/mix.exs [hunk 1]
```diff
@@ -5,8 +5,7 @@ defmodule RustlerTest.Mixfile do
     [
       app: :rustler_test,
       version: "0.0.1",
-      elixir: "~> 1.2",
-      compilers: [:rustler] ++ Mix.compilers(),
```
Comment (evnu) [inline rustler_tests/mix.exs:9]: This seems unrelated to the PR?

Comment (philss) [reply rustler_tests/mix.exs:9]: Sorry, it is not related. Reverted.

**Review by evnu — APPROVED** [review]
